File: 10828.js
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().split('\r\n');
input.shift();

let stack = [];
let result = "";

input.forEach(order => {
  let [cmd, opt] = order.split(' ');
  
  switch (cmd) {
    case "push":
      stack.push(opt);
      break;
    case "top":
      result += `${stack.length == 0 ? -1 : stack[stack.length-1]}\n`;
      break;
    case "size":
      result += `${stack.length}\n`;
      break;
    case "empty":
      result += `${stack.length == 0 ? 1 : 0}\n`;
      break;
    case "pop":
      result += `${stack.length == 0 ? -1 : stack.pop()}\n`;
      break;
  }
});

console.log(result);

File: 10828.cpp
#include <bits/stdc++.h>
using namespace std;


int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	
	stack<int> numStack;
	int n;
	
	cin >> n;
	
	while(n--) {
		string c;
		cin >> c;
		if (c == "push") {
			int tmp;
			cin >> tmp;
			numStack.push(tmp);
		} else if (c == "top") {
			if (numStack.empty())
				cout << -1 << '\n';
			else {
				cout << numStack.top() << '\n';
			}
		} else if (c == "size") {
			cout << numStack.size() << '\n';	
		} else if (c == "pop") {
			if (numStack.empty())
				cout << -1 << '\n';
			else {
				cout << numStack.top() << '\n';
				numStack.pop();	
			}
		} else if (c == "empty") {
			cout << (int)numStack.empty() << '\n';	
		}		
		
	}
	
	
	return 0;
}